src/handleLoginCliCommand.ts

```diff
@@ -1,22 +1,57 @@
 import type { PrinterBlueprint } from './printer.js';
 import { openURL } from './utils.js';
+import { writeFile } from 'node:fs/promises';
+import { homedir } from 'node:os';
+import { join } from 'node:path';
+import { validateAccessToken } from './apis.js';
 
 const ACCESS_TOKEN_REQUESTED_BY_CLI_KEY = 'accessTokenRequestedByCLI';
 
-export const handleLoginCliCommand = async (printer: PrinterBlueprint) => {
-	printer.printConsoleMessage(
-		'info',
-		'Opening the Codemod Studio... Please Sign in with Github.',
-	);
-	const success = openURL(
-		`https://codemod.studio/${ACCESS_TOKEN_REQUESTED_BY_CLI_KEY}`,
-	);
-	if (!success) {
+export const handleLoginCliCommand = async (
+	printer: PrinterBlueprint,
+	token: string | null,
+) => {
+	if (token === null) {
+		printer.printConsoleMessage(
+			'info',
+			'Opening the Codemod Studio... Please Sign in with Github!',
+		);
+		const success = openURL(
+			`https://codemod.studio/${ACCESS_TOKEN_REQUESTED_BY_CLI_KEY}`,
+		);
+		if (!success) {
+			printer.printOperationMessage({
+				kind: 'error',
+				message:
+					'Unexpected error occurred while opening the Codemod Studio.',
+			});
+		}
+		return;
+	}
+
+	const username = await validateAccessToken(token);
+	if (!username) {
 		printer.printOperationMessage({
 			kind: 'error',
 			message:
-				'Unexpected error occurred while opening the Codemod Studio.',
+				'The token is incorrect. Please run `intuita login` again and sign in again in the Codemod Studio.',
 		});
-		return;
 	}
+
+	const globalStoragePath = join(homedir(), '.intuita');
+
+	const buildConfigPath = join(globalStoragePath, 'accessToken.json');
+
+	await writeFile(
+		buildConfigPath,
+		JSON.stringify({
+			accessToken: token,
+			username,
```

Comment:
I think we should always get the username from the token if we need it.

Reply:
addressed